server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -502,7 +506,8 @@ private static String indexMetadataFileName(IndexMetadata indexMetadata) {
         return String.join(
             DELIMITER,
             INDEX_METADATA_FILE_PREFIX,
-            String.valueOf(indexMetadata.getVersion()),
+            RemoteStoreUtils.invertLong(indexMetadata.getVersion()),
+            RemoteStoreUtils.invertLong(INDEX_METADATA_CODEC_VERSION),
```

Comment:
Should we add/update some unit tests for this change ?

Reply:
Are we not using codec version while reading before parsing. We might need to add the format change BWC?

Reply:
There was not any UTs for it, so no change needed. 

Although I have added new UTs for verifying file names.

Reply:
> Are we not using codec version while reading before parsing. We might need to add the format change BWC?

The manifest file parsing need to handle BWC with the introduction of global metadata